packages/leap/lib/src/leap_game.dart

```diff
@@ -1,58 +1,57 @@
 import 'dart:ui';
 
+import 'package:flame/components.dart';
 import 'package:flame/game.dart';
-import 'package:leap/leap.dart';
-import 'package:leap/src/utils/has_tracked_components.dart';
-import 'package:leap/src/utils/lifecycle_state_aware.dart';
+import 'package:leap/src/entities/entities.dart';
+import 'package:leap/src/leap_map.dart';
+import 'package:leap/src/leap_world.dart';
+import 'package:leap/src/mixins/mixins.dart';
 
-/// A Flame game with all the Leap built-ins
+/// A [FlameGame] with all the Leap built-ins.
 class LeapGame extends FlameGame with HasTrackedComponents {
-  late LeapMap map;
-  late LeapWorld world;
+  LeapGame() : appState = AppLifecycleState.resumed;
 
-  AppLifecycleState appState = AppLifecycleState.resumed;
+  late final LeapMap map;
+  late final LeapWorld world;
+  late final CameraComponent cameraComponent;
+  AppLifecycleState appState;
 
   @override
   void lifecycleStateChange(AppLifecycleState state) {
     super.lifecycleStateChange(state);
-
     final oldAppState = appState;
     appState = state;
     for (final c in children.query<AppLifecycleAware>()) {
       c.appLifecycleStateChanged(oldAppState, state);
     }
   }
 
-  /// Tile size (width and height) in pixels
+  /// Tile size (width and height) in pixels.
   double get tileSize => world.tileSize;
 
-  /// All the physical entities in the world
+  /// All the physical entities in the world.
   Iterable<PhysicalEntity> get physicals => world.physicals;
 
   /// Initializes and loads the [world] and [map] components with a Tiled map,
   /// the map file is loaded from "assets/tiled/[tiledMapPath]", and should
-  /// use tile size [tileSize]
-  Future<void> loadWorldAndMap(String tiledMapPath, double tileSize) async {
```

Comment:
I have a partial fix for the camera, we need to use the Flame `World` to get all the proper fixed aspect ratio and follow behavior:

```dart
  /// Initializes and loads the [world] and [map] components with a Tiled map,
  /// the map file is loaded from "assets/tiled/[tiledMapPath]", and should
  /// use tile size [tileSize].
  Future<void> loadWorldAndMap({
    required String tiledMapPath,
    required double tileSize,
    required int tileCameraWidth,
    required int tileCameraHeight,
  }) async {
    final flameWorld = World();
    await add(flameWorld);

    // Default the camera size to the bounds of the Tiled map.
    cameraComponent = CameraComponent.withFixedResolution(
      width: tileSize * tileCameraWidth,
      height: tileSize * tileCameraHeight,
      world: flameWorld,
    );
    await add(cameraComponent);

    // These two classes reference each other, so the order matters here to
    // load properly.
    world = LeapWorld(tileSize: tileSize);
    map = await LeapMap.load('map.tmx', tileSize);

    await flameWorld.addAll([map, world]);
  }

```